tests/test_cookiecutter.py

```diff
@@ -12,211 +13,192 @@
-            self.pip_install(zsh=True)
-
-        # install package and check correct install
-        result = subprocess.Popen(
-            f"pip show {config_dict['package_name']}",
-            shell=True,
-            stdout=subprocess.PIPE,
-        )
-        stdout, __ = result.communicate()
-
-        # Home-page is not in pip show output...
-        show_details = stdout.decode("utf8")
-        assert "Name: test-cookiecutter" in show_details
-        assert "Version: 0.1.dev0" in show_details
-        assert "Summary: A simple Python package" in show_details
-        assert (
-            "Author-email: Test Cookiecutter <testing@cookiecutter.com>"
-            in show_details
-        )
-        assert "License: BSD-3-Clause" in show_details
-
-    def setup_paths(self):
-        config_dict = self.load_configs()
-        package_path = TEST_DIR / config_dict["package_name"]
-        return config_dict, package_path
-
-    @staticmethod
-    def pip_install(zsh=False):
-        if zsh:
-            cmd = "pip install -e '.[dev]'"
-        else:
-            cmd = "pip install -e .[dev]"
-        result = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
-        stdout, __ = result.communicate()
-        return stdout
-
-
-tester = TestCookieCutter()
-tester.setup_cookiecutter()
-tester.test_directory_names()
-tester.check_pyproject_toml()
-tester.test_pip_install()
+with open(f"{CONFIG_FILENAME}", "r") as config_file:
+    config_dict = yaml.full_load(config_file)
+
+config_dict = config_dict["default_context"]
+
+
+@pytest.fixture(autouse=True, scope="session")
+def package_path(tmp_path_factory):
+    # Runs cookiecutter in a temporary directory, and returns
+    # that directory.
+    tmp_path = tmp_path_factory.mktemp("package")
+    os.chdir(tmp_path)
+    subprocess.run(
+        f"cookiecutter {COOKIECUTTER_DIR} --config-file "
+        f"{CONFIG_FILENAME} --no-input",
+        shell=True,
+    )
+
+    return tmp_path / config_dict["package_name"]
+
+
+@pytest.fixture(autouse=True, scope="session")
+def pip_install(package_path):
+    # Installs the package using pip
+    os.chdir(package_path)
+
+    # Uninstall and check package not installed
+    subprocess.run("git init", shell=True)
+    uninstall_cmd = f"pip uninstall -y {config_dict['package_name']}"
+    subprocess.run(uninstall_cmd, shell=True)
+    result = subprocess.Popen(
+        f"pip show {config_dict['package_name']}",
+        shell=True,
+        stderr=subprocess.PIPE,
+    )
+    __, stderr = result.communicate()
+    assert (
+        f"WARNING: Package(s) not found: "
+        f"{config_dict['package_name']}" in stderr.decode("utf8")
+    )
+
+    # Install package
+    dev_formatting = ".[dev]" if platform.system() == "Windows" else "'.[dev]'"
+    cmd = f"pip install -e {dev_formatting}"
+    result = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+    stdout, __ = result.communicate()
```

Comment:
could this line be removed?

Reply:
Looks like it, I've cleaned this up a bit in the most recent commit